userspace/libscap/engine/gvisor/scap_gvisor.cpp

```diff
@@ -86,33 +93,63 @@ engine::~engine()
 
 }
 
-int32_t engine::init(std::string socket_path)
+int32_t engine::init(bool generate_config, std::string socket_path, std::string runsc_root_path, std::string trace_session_config_path)
 {
-	// Initialize the listen fd
-	int sock, ret;
-	struct sockaddr_un address;
-	unsigned long old_umask;
+	if(generate_config)
+	{
+		print_trace_session_config();
+		return SCAP_EOF;
+	}
```

Comment:
I don't think this is the right spot for this call :/
When using Falco, we don't want any initialization to happen before that function is called. Otherwise we would get in output a bunch of log lines and then the json file which is not what we want. I can see two ways out of this:
1. Add it as a separate exported function not part of the engine class (that only exists if gvisor is compiled, of course) which is wired all the way through scap and sinsp
2. Add it straight to sinsp

However, since @Andreagit97 already approved the rest of this PR I would suggest to factor this out of this PR and make another one just for this as the two are independent. WDYT?